src/main/java/net/fabricmc/loom/configuration/processors/MinecraftProcessedProvider.java

```diff
@@ -37,15 +39,20 @@
 import net.fabricmc.loom.util.Constants;
 
 public class MinecraftProcessedProvider extends MinecraftMappedProvider {
-	public static final String PROJECT_MAPPED_CLASSIFIER = "projectmapped";
+	public final String projectMappedClassifier;
 
 	private File projectMappedJar;
 
 	private final JarProcessorManager jarProcessorManager;
 
+	private static String getProjectUuid(Project project) {
+		return UUID.nameUUIDFromBytes(project.getProjectDir().toString().getBytes(StandardCharsets.UTF_8)).toString();
```

Comment:
Im not sure if a whole UUID is needed? would just the project name (formatted in a classifier safe format) be a better solution?

Reply:
In theory, anything that unique for the project should be fine. UUID is just for naming collision avoidance. I don't know if the project name would be safe enough for this. Can people define two sub-projects with the same name?

Reply:
I think if you use full subproject names, you are fine. You can define like `:sub1:duplicate` and `:sub2:duplicate` subprojects within one project I think, but their full project names differ.

Reply:
I see, I will change it to the fully-qualified-name. It will be more human-readable, and hopefully shorter. A commit will be up for this tomorrow.

Reply:
A commit is up. My biggest concern is what if there is invalid char (i.e. cannot exists in the path) inside the project name. Any solution for this? Should we write a transformer that replaces all the invalid char?

I cannot find a good resource for what is a valid name for the Gradle project, seems and char except `:` are legal.